tests/test_mesos_checkpointing.py

```diff
@@ -0,0 +1,124 @@
+import sdk_networks
+import sdk_tasks
+import shakedown
+import spark_utils as utils
+
+log = logging.getLogger(__name__)
+
+start_agent_cmd = "sudo systemctl start dcos-mesos-slave"
+stop_agent_cmd = "sudo systemctl stop dcos-mesos-slave"
+check_agent_cmd = "sudo systemctl is-active dcos-mesos-slave"
+
+app_name = "MockTaskRunner"
+driver_cpus = None
+executor_cpus = None
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_spark(configure_security_spark, configure_universe):
+    try:
+        utils.upload_dcos_test_jar()
+        utils.require_spark()
+
+        # We need to pick two nodes with the maximum unused CPU to guarantee that Driver and Executor
+        # are not running on the same host.
+        available_cpus = []
+        cluster_agents = sdk_cmd.cluster_request('GET', '/mesos/slaves').json()
+        for agent in cluster_agents["slaves"]:
+            available_cpus.append(int(float(agent["resources"]["cpus"])) - int(float(agent["used_resources"]["cpus"])))
+
+        available_cpus.sort(reverse=True)
+        assert len(available_cpus) >= 3, \
+            "Expected 3 or more nodes in the cluster to accommodate Dispatcher, " \
+            "Driver, and Executor each on a separate node"
+
+        global driver_cpus
+        driver_cpus = available_cpus[0]
+
+        global executor_cpus
+        executor_cpus = available_cpus[1]
+
+        log.info(f"{driver_cpus} cores will be used for driver, {executor_cpus} cores will be used for executor")
+        yield
+    finally:
+        utils.teardown_spark()
+
+
+@pytest.mark.sanity
+def test_agent_restart_with_checkpointing_disabled():
+    (driver_task, executor_task) = next(_submit_job_and_get_tasks())
+
+    driver_ip = sdk_networks.get_task_host(driver_task)
+    executor_ip = sdk_networks.get_task_host(driver_task)
+
+    # Dispatcher starts Driver Tasks with checkpointing enabled so Driver is expected to be in RUNNING state
+    _restart_task_agent_and_verify_state(driver_ip, driver_task, "TASK_RUNNING")
+    _restart_task_agent_and_verify_state(executor_ip, executor_task, "TASK_KILLED")
+
+
+@pytest.mark.sanity
+def test_agent_restart_with_checkpointing_enabled():
+    (driver_task, executor_task) = next(_submit_job_and_get_tasks(extra_args=["--conf spark.mesos.checkpoint=true"]))
+
+    driver_ip = sdk_networks.get_task_host(driver_task)
+    executor_ip = sdk_networks.get_task_host(driver_task)
+
+    _restart_task_agent_and_verify_state(executor_ip, executor_task, "TASK_RUNNING")
+    _restart_task_agent_and_verify_state(driver_ip, driver_task, "TASK_RUNNING")
+
+
+def _restart_task_agent_and_verify_state(host_ip, task, expected_state):
+    dcos_utils.agent_ssh(host_ip, stop_agent_cmd)
+    _check_agent_status(host_ip, "inactive")
+    dcos_utils.agent_ssh(host_ip, start_agent_cmd)
+    _check_agent_status(host_ip, "active")
+    _wait_for_task_status(task["id"], expected_state)
+
+
+@retrying.retry(
+        wait_fixed=5000,
+        stop_max_delay=120 * 1000,
```

Comment:
is 2 minutes enough time for Mesos agent to come back to life?

Reply:
We're checking for agent status separately for stop and then for start. So we wait for two minutes for agent process to start from a stopped state which I think is plenty of time.

Reply:
yeah I figure stopping within 2 minutes is no issue.. I just wasn't sure about startup but if you think 2 minutes is enough then 👍 from me